data_steward/analytics/cdr_ops/rt_cdr_qc/cdr_deid_qa_report6_fitdata.py

```diff
@@ -358,16 +358,82 @@
 
 result = execute(client, query)
 
-if sum(result['bad_rows']) == 0:    
-    summary = summary.append(        {
-            'query': 'Query7 device_id was deidentified properly for all records.',
+if sum(result['bad_rows']) == 0:
+    summary = summary.append(
+        {
+            'query':
+                'Query7 device_id was deidentified properly for all records.',
+            'result':
+                'PASS'
+        },
+        ignore_index=True)
+else:
+    summary = summary.append(
+        {
+            'query':
+                'Query7 device_id was not deidentified properly. See query description for hints.',
+            'result':
+                'Failure'
+        },
+        ignore_index=True)
+result
+
+# -
+
+# # Check de-identification of src_ids.
+#
+# DC-3376
+#
+# This check verifies that src_ids in fitbit tables were updated successfully. The check passes if no records are returned.
+
+# +
+src_id_check = JINJA_ENV.from_string("""
+SELECT
+    '{{table}}' as table,
+    ft.person_id,
+    COUNT(1) bad_rows
+FROM
+    `{{project}}.{{deid_cdr_fitbit}}.{{table}}` ft
+JOIN
+    `{{project}}.{{non_deid_fitbit}}.{{table}}` st
+ON
+    ft.person_id = st.person_id
+JOIN
+    `{{project}}.{{pipeline_tables}}.{{site_maskings}}` sm
+ON
+    sm.hpo_id = st.src_id
+WHERE
+  ft.src_id != sm.src_id
+OR
+  ft.src_id is NULL
+GROUP BY 
+    person_id
+""")
```

Comment:
To properly join ft and st you'd need to join on all fields as there are no primary keys for fitbit. If you only join on person_id you'd be making quite a few duplicates because participants could have many rows in each table.
However, I don't think you need to join to the {non_deid_fitbit} table at all. This query could look in the deid table for rows that don't have the expected values correct?

Reply:
Thanks for the changes!